python/cudf/cudf/api/types/categoricals.py

```diff
@@ -1,40 +0,0 @@
-# Copyright (c) 2021, NVIDIA CORPORATION.
-
-from __future__ import annotations
-
-from typing import List, Union
-
-import cudf
-
-ParentType = Union["cudf.Series", "cudf.Index"]
-
-
-def _union_categoricals(
-    to_union: List[Union[cudf.Series, cudf.Index]],
-    sort_categories: bool = False,
-    ignore_order: bool = False,
-):
-    """
-    This is an internal API which combines categorical data.
-    """
-    # TODO(s) in the order specified :
-    # 1. The return type needs to be changed
-    #    to cudf.Categorical once it is implemented.
-    # 2. Make this API public (i.e., to resemble
-    #    pd.api.types.union_categoricals)
-
-    if ignore_order:
-        raise TypeError("ignore_order is not yet implemented")
-
-    result_col = cudf.core.column.CategoricalColumn._concat(
-        [obj._column for obj in to_union]
-    )
-    if sort_categories:
-        sorted_categories = result_col.categories.sort_by_values(
-            ascending=True
-        )[0]
-        result_col = result_col.cat().reorder_categories(
-            new_categories=sorted_categories
-        )
-
-    return cudf.Index(result_col)
```

Comment:
Would ask @galipremsagar if moving this utility is OK as I believe something in Dask/dask-cudf depends on this.

Reply:
This refactor should be okay with respect to `dask-cudf` since this API was also previously available at `cudf.api.types._union_categoricals`

Reply:
Yep I think this is fine, I made sure to match the namespace that this function was being pulled from. The main change is that types is now a single module rather than a subpackage with separate modules for different types. We can always revisit that organization in the future if we prefer, but I don't the subpackage is necessary.